samples/snippets/run_report_with_filters.py

```diff
@@ -0,0 +1,212 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Google Analytics Data API sample application demonstrating the usage of
+dimension and metric filters in a report.
+"""
+from google.analytics.data_v1beta import BetaAnalyticsDataClient
+from google.analytics.data_v1beta.types import DateRange
+from google.analytics.data_v1beta.types import Dimension
+from google.analytics.data_v1beta.types import Filter
+from google.analytics.data_v1beta.types import FilterExpression
+from google.analytics.data_v1beta.types import FilterExpressionList
+from google.analytics.data_v1beta.types import Metric
+from google.analytics.data_v1beta.types import NumericValue
+from google.analytics.data_v1beta.types import RunReportRequest
+
+from run_report import print_run_report_response
+
+
+def run_sample():
+    """Runs the sample."""
+    # TODO(developer): Replace this variable with your Google Analytics 4
+    #  property ID before running the sample.
+    property_id = "YOUR-GA4-PROPERTY-ID"
+    run_report_with_dimension_filter(property_id)
+    run_report_with_multiple_dimension_filters(property_id)
+    run_report_with_dimension_exclude_filter(property_id)
+    run_report_with_dimension_in_list_filter(property_id)
+    run_report_with_dimension_and_metric_filters(property_id)
+
+
+def run_report_with_dimension_filter(property_id="YOUR-GA4-PROPERTY-ID"):
+    """Runs a report using a dimension filter. The call returns a time series
+    report of `eventCount` when `eventName` is `first_open` for each date."""
+    client = BetaAnalyticsDataClient()
+
+    # [START analyticsdata_run_report_with_dimension_filter]
+    request = RunReportRequest(
+        property="properties/" + str(property_id),
+        dimensions=[Dimension(name="date")],
+        metrics=[Metric(name="eventCount")],
+        date_ranges=[DateRange(start_date="7daysAgo", end_date="yesterday")],
+        dimension_filter=FilterExpression(
+            filter=Filter(
+                field_name="eventName",
+                string_filter=Filter.StringFilter(value="first_open"),
+            )
+        ),
+    )
+    response = client.run_report(request)
+    # [END analyticsdata_run_report_with_dimension_filter]
+    print_run_report_response(response)
+
+
+def run_report_with_multiple_dimension_filters(property_id="YOUR-GA4-PROPERTY-ID"):
+    """Runs a report using multiple dimension filters joined as `and_group`
+    expression. The filter selects for when both `browser` is `Chrome` and
+    `countryId` is `US`."""
+    client = BetaAnalyticsDataClient()
+
+    # [START analyticsdata_run_report_with_multiple_dimension_filters]
+    request = RunReportRequest(
+        property="properties/" + str(property_id),
+        dimensions=[Dimension(name="browser")],
+        metrics=[Metric(name="activeUsers")],
+        date_ranges=[DateRange(start_date="7daysAgo", end_date="yesterday")],
```

Comment:
nit: Missed a date range update

Reply:
Added a link to the date range reference documentation instead.